main.go

```diff
@@ -25,34 +25,19 @@ func Handler(ctx context.Context, snsEvent events.SNSEvent) {
 
 	for _, record := range snsEvent.Records {
 		msg, _ := message.Parse(record)
-		switch msg.Action {
-		case "touch":
-			updateOne(msg, client, indexer)
-		case "delete":
-			rebuildIndex(msg, client, indexer)
-		}
+		actionForMessage(msg, client, indexer).Run(msg)
```

Comment:
I see that `Run()` returns an `error`. Should we be grabbing the return value here and logging something if it fails? (Generally, I want to make sure we're logging errors in our lambdas.)